addons/account/models/account_move.py

```diff
@@ -391,6 +391,20 @@ def _sequence_fixed_regex(self):
         required=True,
         compute='_compute_currency_id', inverse='_inverse_currency_id', store=True, readonly=False, precompute=True,
     )
+    exchange_rate = fields.Float(
+        string="Currency rate",
+        tracking=True,
+        copy=False,
+    )
+    system_exchange_rate = fields.Float(
+        compute='_compute_system_exchange_rate',
+        help="Expected exchange rate based on document invoice date, company, and currency"
+    )
+    display_rate = fields.Float(
+        string="Exchange Rate",
+        compute='_compute_display_rate', inverse='_inverse_display_rate',
+        help="Use this exchange rate instead of the default rate."
+    )
```

Comment:
@h4818 Is not 3 fields a bit much?  You are not using system_exchange_rate in the view, so there is no need for that field.  I had just one stored field in mind which would always have the rate, but somehow I could understand that you want to know when you set it manually or not, so you know what to use when recomputing automatically.

Reply:
Indeed, that's the exact reason: to determine whether a custom rate is used or not
There is only 1 stored field.

**System rate**: computed from res.currency if the move is in draft. It is recomputed every time the date/company changes, therefore it is in the view and invisible.
**Exchange rate**: Stores the custom rate or 0.0 if the system rate is used
**Display rate**: computed from either custom rate or system rate. This is the field displayed on the view allowing a user to set a custom rate, or default to the system rate.